### ISSUE TITLE
High training accuracy (99%) low test\val result ( 60%) 

### ISSUE THREAD

**Issue by h3nok** [Microsoft/CNTK#1529]
Hi, I have a model, much like AlexNet architecture wise with few modification to the convolution layers and the training procedure. The network achieves very high accuracy during training. However, when I test it with unseen set of samples the accuracy stagnates at about 60%. I am trying to understand why this is the case? You would think there would be high correlation between training and validation accuracies. What would you make of this result?  The network is trained on about 70000 images of retina with in-training augmentation. It has 6 outputs. Thanks

**Comment by L-Ramos** [comment]
Hi h3nok, this is usually a problem related to over-fitting, your network is not generalizing properly. Can you tell me what is the size of your input image? Also how many images do you have after augmentation?

**Comment by h3nok** [comment]
Hi L-Ramos, thanks for the replay. Now that you mentioned it, the network does seem to be over-fitting. When I measure its sensitivity it is more sensitive to class 1 than the rest. I started off with about 35K images then after mirroring each image (as a preprocessing) the network takes total of ~70K images. At the end the network sees total of roughly 140000 images. CNTK crop, color jitter transforms are used in the brain script. It took some parameter tuning to get the network to train and converge with 98% accuracy, which I though was good result. Prior to that the network wasn't learning at all. It would stagnate about 61% error rate. What is the best way to battle this when you have limited amount of data? thank you for your time.

**Comment by L-Ramos** [comment]
Hi h3nok, so, you have 35K images, augment it by a factor of 2 and get 70K right? After that do you randomly split the dataset into training and testing right? Like 70/80% to training and the rest for testing. You said you have 6 classes, are they equally balanced? For instance, is it possible that from the 35K images 15K are just from one class? Also what is the size of your image that you feed to the network (like, 50x50, 25x25). Does it contain only retina or do you have other information too, like background, etc...

**Comment by h3nok** [comment]
That is right. After that I use 80/20 split for training and testing. I also made sure the dataset is balanced across all 6 classes. The size of the images are 512x512. Each sample contains retina with black background.

**Comment by L-Ramos** [comment]
Thats a really big image, how much of the image is actually retina and how much is background? It may be the case that its not really learning to distinguish classes, but background from retina. Also 512x512 will give you a lot of parameters to train, millions of them for just a few hundred thousand examples. If you take a look at the AlexNet paper, with a 216x216x3 input they had over 60 million parameters to train, and that requires a lot of that. I suggest the following:
1st : Cropping retina or extract patches from it
2nd : if 1 is not possible, reduce the image scale (256x256 or smaller)
3rd : try a different architecture, maybe a more shallow network is better than a deeper one.

**Comment by h3nok** [comment]
![sample](https://cloud.githubusercontent.com/assets/22085306/23368270/8336b022-fcca-11e6-9c27-63d9f2e4d587.jpg)

Most examples have background to retina ratio just like the images above (these are actually the model images). Good point about it learning to distinguish background from retina. I did not think of that. I do however apply cropping of cropType=center, sideRatio= 0.8 so that the network sees at least less  background compared to retina. I have also tried 256x256 with similar results (i may have to use this again for comparison). I will try to extract patches or the retina  as a way to reduce the background pixels. FIY this is part of my ongoing research and a planned paper that I hope to publish to highlight guidelines for getting reasonable results with small datasets and the sorts of augmentation\ pre-processing steps one uses to boost the networks performance. In that regard shallow networks are not an option. I appreciate your feedback.

**Comment by h3nok** [comment]
![retina_patch](https://cloud.githubusercontent.com/assets/22085306/23579486/a7a306ba-00ab-11e7-9337-21b83a620af3.png)

Looks like excluding the background black pixels from training seems to have boosted the performance. The network now performs at about 81% accuracy on test set. Thanks for your input.

**Comment by L-Ramos** [comment]
Thats really cool, I'm happy to help!

**Comment by Billsbob** [comment]
I have a similar task where the desired area is very small relative to the overall image and training accuracy is over 95% from the first epoch.  After reducing image size and cropping out sections the training accuracy has improved, but not by much.  Is there a way to make the training accuracy more sensitive so that 95% is treated as if it was 50%?  New to ml, so i'm not sure where to look for this?

**Comment by delzac** [comment]
@Billsbob if you are going semantic segmentation then you should use [fmeasure ](https://github.com/Microsoft/CNTK/blob/master/bindings/python/cntk/losses/__init__.py#L424)as your loss.

Alternatively, if you are doing classification then you can consider using focal loss. I have already implemented this in my own cntk extension library called [CNTKx](https://github.com/delzac/cntkx)

**Comment by h3nok** [comment]
This behavior is characteristic of overfitting. Look into ways to enhance your training set and increase the number of samples per class. You can try image augmentation techniques. Not sure CNTK has any image augmentation routines.
I know you can accomplish that with Keras or other libraries.  Below is keras example. Multiply each image by 20 or so using different augmentation techniques and see if your training results change.

Best

https://towardsdatascience.com/image-augmentation-for-deep-learning-using-keras-and-histogram-equalization-9329f6ae5085
